### DISCUSSION TITLE
404 error in production despite the build being successfull

### DISCUSSION THREAD

**Discussion by joshuaolusayo** [vercel/vercel#8750] [Help] [answered]
I am trying to deploy a site. the build was successfull but it is not showing anything in product. It is showing 404 not found.
![image](https://user-images.githubusercontent.com/53523015/196767214-a04f5d4d-4334-424f-ae4c-50ebbf6f0cd9.png)
![image](https://user-images.githubusercontent.com/53523015/196767240-47f2de4b-15a3-48a9-bdec-a6d1aee30e47.png)

**Comment by amyegan** [comment]
Sorry you're running into this problem @joshuaolusayo. Can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) to help me get a better look at what is going wrong?

**Comment by joshuaolusayo** [comment]
Not sure I can because my other apps are running well. It just started
today.

**Comment by Aleksandr-Pod** [comment]
You are not alone. I have the same problem. And a month before my friend had the same issue.
And it's weirdly that no any clear explanation here/

**Comment by joshuaolusayo** [comment]
I really hope we get solution soon

**Comment by andrew55Works** [comment]
Same issue here, I did not change the project setting but after new pushing this afternoon, it just appears 404 page.
I like Vercel and even pay for the team plan. Hope we could get a solution. Our clients don't wait.

**Comment by amyegan** [comment]
I'm sorry so many of you are running into this problem. There are many possible causes, so I need more information to help with debugging.

If you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) I would be happy to take a closer look!

**Comment by kiran-blip** [comment]
The minimal repro examples are above. I deployed a Next.js app, have one env variable and saw this page.

**Reply by amyegan** [reply]
Maybe I'm just missing it, but I can't find any minimal reproductions shared by anyone here yet. 😕 I haven't been able to see the problem in my own example site. Do you have a minimal repro that you can link here for me to use for debugging?

**Comment by webpro** [comment]
Perhaps I'm having the same issue. I've opened a case for this. Here is the repo. It builds successfully, but gives a 404 at the root. Nothing fancy going on, default settings all the way down: https://github.com/7-docs/demo-next

**Reply by amyegan** [reply]
I tried deploying that example, but I don't get a 404 at the root. 

- Deployment URL: https://demo-next-ashy-two.vercel.app/
- Build & Deployment Settings: `Next.js` framework preset, no overrides
- Root Directory: blank by default
- Node.js Version: `18.x`

The only error I get is a 500 on `/api/completion`, which I expected because I didn't set up any environment variables. Can you share screenshots or walk me through the steps I should take to see the 404?

**Reply by webpro** [reply, marked as answer]
Apologies for not following up. I got a reply from your colleague Joshua which helped me out, everything is fine now. There were two issues:

- During setup, there was a Next.js icon next to the GitHub repo to connect, so I didn't think about setting the Next.js framework preset, it was still unset or "custom". This was the main issue, it didn't show any errors.
- During the app init from the command line, there was an option to use `./src` as the source folder, which I used. So I also set that as the "root directory" in the GUI when connecting the repo. This was the second mistake.

Those were mistakes on my end, but I think both can be prevented with some checks.
